CuDF backend doesn't check CuDF version before doing version-dependent things

I had cudf 24.04.01 installed (apparently, when a package comes from another channel, `update --all` doesn't get it), and got this error from `tests-cuda/test_3051_to_cuda.py`:

```
>       return cudf.core.column.lists.ListColumn(
            size=length,
            data=None,
            mask=m,
            children=(ind_buf, cont),
            dtype=cudf.core.dtypes.ListDtype(cont.dtype),
        )
E       TypeError: ListColumn.__init__() got an unexpected keyword argument 'data'
E       
E       This error occurred while calling
E       
E           ak.to_cudf(
E               <Array [[[1, 2, 3], [], [3, 4]], []] type='2 * var * var * int64'>
E           )
```

Naturally, this is #3309. I updated cudf to 24.10.01 and there's no problem.

If you try to use this feature without CuDF, I'm sure you'd get some error message saying, "you need to install CuDF," but the above error message is mysterious if you don't know that you have an old version of CuDF. It would be more helpful if it said, "you need to install VERSION XYZ of CuDF," like

https://github.com/scikit-hep/awkward/blob/8f69dbf6eb84f518150f47a81f9605642bbff837/src/awkward/numba/__init__.py#L34-L38

@martindurant, is 24.10.00 the exact minimum? Which is the minimum CuDF version that is okay/the maximum CuDF version that is bad?

--- a/src/awkward/contents/listoffsetarray.py
+++ b/src/awkward/contents/listoffsetarray.py
@@ -2010,6 +2010,7 @@ class ListOffsetArray(ListOffsetMeta[Content], Content):
 
     def _to_cudf(self, cudf: Any, mask: Content | None, length: int):
         from packaging.version import parse as parse_version
+
         cupy = Cupy.instance()
         index = self._offsets.raw(cupy).astype("int32")
         buf = cudf.core.buffer.as_buffer(index)

--- a/src/awkward/contents/listoffsetarray.py
+++ b/src/awkward/contents/listoffsetarray.py
@@ -2015,7 +2015,7 @@ class ListOffsetArray(ListOffsetMeta[Content], Content):
         index = self._offsets.raw(cupy).astype("int32")
         buf = cudf.core.buffer.as_buffer(index)
 
-        if parse_version(cupy._version.__version__) >= parse_version("24.10.00"):
+        if parse_version(cudf.__version__) >= parse_version("24.10.00"):
             ind_buf = cudf.core.column.numerical.NumericalColumn(
                 data=buf, dtype=index.dtype, mask=None, size=len(index)
             )
@@ -2042,13 +2042,21 @@ class ListOffsetArray(ListOffsetMeta[Content], Content):
                 mask=m,
             )
 
-        return cudf.core.column.lists.ListColumn(
-            size=length,
-            data=None,
-            mask=m,
-            children=(ind_buf, cont),
-            dtype=cudf.core.dtypes.ListDtype(cont.dtype),
-        )
+        if parse_version(cudf.__version__) >= parse_version("24.10.00"):
+            return cudf.core.column.lists.ListColumn(
+                size=length,
+                data=None,
+                mask=m,
+                children=(ind_buf, cont),
+                dtype=cudf.core.dtypes.ListDtype(cont.dtype),
+            )
+        else:
+            return cudf.core.column.lists.ListColumn(
+                length,
+                mask=m,
+                children=(ind_buf, cont),
+                dtype=cudf.core.dtypes.ListDtype(cont.dtype),
+            )
 
     def _to_backend_array(self, allow_missing, backend):
         array_param = self.parameter("__array__")

--- a/src/awkward/contents/listoffsetarray.py
+++ b/src/awkward/contents/listoffsetarray.py
@@ -2009,12 +2009,19 @@ class ListOffsetArray(ListOffsetMeta[Content], Content):
             )
 
     def _to_cudf(self, cudf: Any, mask: Content | None, length: int):
+        from packaging.version import parse as parse_version
         cupy = Cupy.instance()
         index = self._offsets.raw(cupy).astype("int32")
         buf = cudf.core.buffer.as_buffer(index)
-        ind_buf = cudf.core.column.numerical.NumericalColumn(
-            data=buf, dtype=index.dtype, mask=None, size=len(index)
-        )
+
+        if parse_version(cupy._version.__version__) >= parse_version("24.10.00"):
+            ind_buf = cudf.core.column.numerical.NumericalColumn(
+                data=buf, dtype=index.dtype, mask=None, size=len(index)
+            )
+        else:
+            ind_buf = cudf.core.column.numerical.NumericalColumn(
+                buf, index.dtype, None, size=len(index)
+            )
         cont = self._content._to_cudf(cudf, None, len(self._content))
         if mask is not None:
             m = np._module.packbits(mask, bitorder="little")

